src/transaction/index.ts

```diff
@@ -1181,14 +1185,35 @@ export class Transaction {
 		if (receiveAddress.isErr()) {
 			return err(receiveAddress.error.message);
 		}
+
+		// try to calculate satsPerByte if not provided.
+		// child + parent combined fee rate should be higher than fastest.
+		if (!satsPerByte && txid) {
+			const parent = this._wallet.data.transactions[txid];
+			if (parent) {
+				const parentVsize = parent.vsize;
+				const childVsize = 141; // assume segwit 1 input 1 output
+				const fast = this._wallet.feeEstimates.fast;
+				const res = Math.ceil(
+					(fast * (parentVsize + childVsize) - parent.fee) / childVsize
+				);
+				satsPerByte = res;
+			}
+		}
+
+		// if we still don't have a satsPerByte, use 1.5x fastest.
+		if (!satsPerByte) {
```

Comment:
Can we make `txId` required? I don't see a case where you don't know which transaction you are trying to boost. Probably still need this code in case the transaction can't be found for some reason but should be rare.

Reply:
At this point, I don't want to interfear much and change function signatures.